core/src/utils/menu-controller/index.ts

```diff
@@ -76,14 +78,31 @@ const createMenuController = () => {
     if (menu === 'start' || menu === 'end') {
       // there could be more than one menu on the same side
       // so first try to get the enabled one
-      const menuRef = find((m) => m.side === menu && !m.disabled);
-      if (menuRef) {
-        return menuRef;
+      const menuRefs = menus.filter((m) => m.side === menu && !m.disabled);
+      if (menuRefs.length >= 1) {
+        if (menuRefs.length > 1) {
+          printIonWarning(
+            `menuController queried for a menu on the "${menu}" side, but ${menuRefs.length} menus were found. The first menu reference will be used. If this is not the behavior you want then pass the ID of the menu instead of its side.`,
```

Comment:
Optional: When I try setting the first `ion-menu-button` in the new `multiple` test to `menu="start"`, I get five copies of this warning in the console immediately on loading/refreshing the page. Would it be easy enough to only log the warning when the menu is actually opened? (If this isn't trivial, don't worry about it.)

Reply:
Good call! I updated this in https://github.com/ionic-team/ionic-framework/pull/28269/commits/0787dc8e61948895ac4570d00b06e052c0739fb8. This only warns when opening a menu.

Reply:
I'm no longer getting the warnings at all, even when opening the menu. I also noticed that with the two menu buttons set to `menu="start"` and `menu="secondary-menu"` respectively, both open the secondary menu, which doesn't seem right; shouldn't it be the primary one since that's first? I think I remember the `menu="start"` button opening the primary one before 🤔

Reply:
The original intent of this was to only warn when users are explicitly calling `menuController.open('side')`, but I can update it so it gets called when `toggle` is called on the menuController (what the menu button does when you click it)

Reply:
Fixed in https://github.com/ionic-team/ionic-framework/pull/28269/commits/01f4a272e2b93ce4fb92f14e557c7c85e98c6d8d